test/gurka/test_admin_sidewalk_override.cc

```diff
@@ -0,0 +1,196 @@
+                  std::set<std::string>& countries,
+                  std::set<std::string>& states) {
+  countries.clear();
+  states.clear();
+
+  // Load the admin.sqlite table we just created
+  sqlite3* db_handle = NULL;
+  uint32_t ret = sqlite3_open_v2(dbname.c_str(), &db_handle, SQLITE_OPEN_READONLY, NULL);
+  EXPECT_EQ(ret, SQLITE_OK);
+
+  sqlite3_stmt* stmt = 0;
+  std::string sql = "SELECT admin_level, name from admins;";
+
+  uint32_t result = 0;
+  bool dor = true;
+  bool intersection_name = false;
+  ret = sqlite3_prepare_v2(db_handle, sql.c_str(), sql.length(), &stmt, 0);
+
+  if (ret == SQLITE_OK || ret == SQLITE_ERROR) {
+    result = sqlite3_step(stmt);
+
+    if (result == SQLITE_DONE) {
+      sqlite3_finalize(stmt);
+      stmt = 0;
+      return;
+    }
+  }
+
+  while (result == SQLITE_ROW) {
+    int admin_level = 0;
+    if (sqlite3_column_type(stmt, 0) == SQLITE_INTEGER)
+      admin_level = sqlite3_column_int(stmt, 0);
+    EXPECT_TRUE((admin_level == 4) || (admin_level == 2));
+
+    std::string name;
+    if (sqlite3_column_type(stmt, 1) == SQLITE_TEXT)
+      name = (char*)sqlite3_column_text(stmt, 1);
+    EXPECT_TRUE(!name.empty());
+
+    if (admin_level == 4)
+      states.insert(name);
+    else
+      countries.insert(name);
+
+    result = sqlite3_step(stmt);
+  }
+
+  if (stmt) {
+    sqlite3_finalize(stmt);
+    stmt = 0;
+  }
+}
+
+} // anonymous namespace
+
+TEST(AdminTest, TestBuildAdminFromPBF) {
+  //======================================================================
+  // part I: create a mock graph, build a pbf from it, build a sqlite
+  // from that pbf, read a few bits from the sqlite to prove its there
+  // and has the things we'd expect.
+  //======================================================================
+
+  // Create test/data/admin/map.pbf
+  const std::string workdir = "test/data/admin_uk";
+
+  if (!filesystem::exists(workdir)) {
+    bool created = filesystem::create_directories(workdir);
+    EXPECT_TRUE(created);
+  }
+
+  valhalla::gurka::map admin_map = BuildPBF(workdir);
+
+  boost::property_tree::ptree& pt = admin_map.config;
+  pt.put("mjolnir.concurrency", 1);
+  pt.put("mjolnir.id_table_size", 1000);
+  pt.put("mjolnir.tile_dir", workdir + "/tiles");
+  pt.put("mjolnir.admin", workdir + "/admin.sqlite");
+  pt.put("mjolnir.timezone", workdir + "/not_needed.sqlite");
+
+  std::string dbname = pt.get<std::string>("mjolnir.admin");
+
+  // Given map.pbf, BuildAdminFromPBF() creates test/data/admin.sqlite.
+  std::vector<std::string> input_files = {workdir + "/map.pbf"};
+  BuildAdminFromPBF(pt.get_child("mjolnir"), input_files);
+
+  // Load the sqlite and read the countries/states from the admin table
+  std::set<std::string> countries, states;
+  GetAdminData(dbname, countries, states);
+
+  std::set<std::string> exp_countries = {"Belarus"};
+  EXPECT_EQ(countries, exp_countries);
+
+  std::set<std::string> exp_states = {};
+  EXPECT_EQ(states, exp_states);
```

Comment:
nit: if you wanted to you could leave this out since its more testing the admin builder than the overrides and we have another test for admin builder anyway. feel free to ignore me though, just saying you went above and beyond :D

Reply:
i left in case ppl wanted to use as a template in the future for other country override tests